### PR TITLE
fix: create real classes instances of entities, read models and events in the Booster class

### PR DESCRIPTION
**Pull request by Moneyba**
## Description
This PR fixes https://github.com/boostercloud/booster/issues/849

## Changes

- Create proper instances for  entities, read models and events in the Booster class.
- Remove the deprecated method `fetchEntitySnapshot`

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [ ] Updated documentation accordingly

### CODE REVIEW

File: packages/framework-core/test/booster.test.ts [hunk 1]
```diff
@@ -81,35 +72,145 @@ describe('the `Booster` class', () => {
...
+          } as EventInterface,
+        },
+      ]
+      const providerEventsSearch = fake.returns(searchResult)
+      Booster.configureCurrentEnv((config) => {
+        config.provider = {
+          events: {
+            search: providerEventsSearch,
+          },
+        } as unknown as ProviderLibrary
```
Comment (juanjoman) [inline packages/framework-core/test/booster.test.ts:130]: is this double `as` on purpose? 😮

File: packages/framework-core/test/booster.test.ts [hunk 2]
```diff
@@ -81,35 +72,145 @@ describe('the `Booster` class', () => {
...
 
-    expect(Booster.serveGraphQL).to.have.been.calledOnceWith(message)
+      expect(Booster.serveGraphQL).to.have.been.calledOnceWith(message)
+    })
+  })
+
+  describe('The `entity` method', () => {
+    afterEach(() => {
+      restore()
+    })
```
Comment (juanjoman) [inline packages/framework-core/test/booster.test.ts:181]: This is at the top of the file too, is it necessary?

File: packages/framework-core/test/booster.test.ts [hunk 3]
```diff
@@ -81,35 +72,145 @@ describe('the `Booster` class', () => {
...
+
+    context('given a BoosterConfig', () => {
+      const config = new BoosterConfig('test')
+      config.provider = {} as ProviderLibrary
+
+      it('the `entity` function calls to the `fetchEntitySnapshot` method in the EventStore', async () => {
+        replace(EventStore.prototype, 'fetchEntitySnapshot', fake.returns({ id: '42' }))
```
Comment (juanjoman) [inline packages/framework-core/test/booster.test.ts:188]: Just to confirm, this `fetchEntitySnapshot` is not the one we wanted to deprecate right? Is it the one from providers?

**Review by juanjoman — APPROVED** [review]
LGTM!! Just left some comments but nothing big, nice work 🎉

File: packages/framework-common-helpers/src/instances.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  Object.assign(instance, rawObject)
+  return instance
+}
+
+export function createInstances<T>(instanceClass: Class<T>, rawObjects: Array<Record<string, any>>): T[] {
+  return rawObjects.map((rawObject) => {
+    const instance = new instanceClass()
+    Object.assign(instance, rawObject)
+    return instance
+  })
```
Comment (alvaroloes) [inline packages/framework-common-helpers/src/instances.ts:14]: What about reusing the above method?
```suggestion
  return rawObjects.map((rawObject) => createInstance(instanceClass, rawObject))
```

File: packages/framework-core/test/booster.test.ts (hunk 1 above)
Comment (alvaroloes) [reply packages/framework-core/test/booster.test.ts:130]: I think it is the only way to bypass the compiler in this case

File: packages/framework-core/test/booster.test.ts [hunk 5]
```diff
@@ -81,35 +72,141 @@ describe('the `Booster` class', () => {
...
+
+      for (const event of events) {
+        let eventValue
+        switch (event.type) {
+          case TestEvent.name:
+            eventValue = event.value as TestEvent
+            expect(eventValue.getId()).to.not.throw
+            break
+          default:
+            console.log('Default')
```
Comment (alvaroloes) [inline packages/framework-core/test/booster.test.ts:147]: Is this log required?

File: packages/framework-core/test/booster.test.ts [hunk 6]
```diff
@@ -81,35 +72,141 @@ describe('the `Booster` class', () => {
...
+          type: TestEvent.name,
+          entity: random.alpha(),
+          entityID: random.uuid(),
+          createdAt: random.alphaNumeric(),
+          value: {
+            id: '1',
+            entityID: () => UUID.generate(),
+          } as EventInterface,
+        },
+      ]
```
Comment (alvaroloes) [inline packages/framework-core/test/booster.test.ts:123]: Would you mind including another raw event of another type to test that an array of events of different types is correctly handled, please?

**Review by alvaroloes — COMMENTED** [review]
Left some comments, but the rest LGTM!

File: packages/framework-core/test/booster.test.ts (hunk 1 above)
Comment (Moneyba) [reply packages/framework-core/test/booster.test.ts:150]: Yes, that's it. The other option would be as `any`. Thanks @alvaroloes!

File: packages/framework-core/test/booster.test.ts (hunk 3 above)
Comment (Moneyba) [reply packages/framework-core/test/booster.test.ts:207]: `fetchEntitySnapshot` is a function inside of the class `EventStore`.

**Review by alvaroloes — APPROVED** [review]

File: packages/framework-core/test/booster.test.ts (hunk 1 above)
Comment (juanjoman) [reply packages/framework-core/test/booster.test.ts:150]: Awesome! 😄

**Pull request merged**